### ISSUE TITLE
AttributeError: 'str' object has no attribute 'hex'

### ISSUE THREAD

**Issue by karlb** [raiden-network/raiden#6943]
When running the pfs3 scenario locally on the presence_less_client_rebase branch, I get the following traceback:

```
Traceback (most recent call last):
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/flask_restful/__init__.py", line 468, in wrapper
    resp = resource(*args, **kwargs)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/flask/views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/flask_restful/__init__.py", line 583, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/api/rest_utils.py", line 59, in decorated
    return method(self, *args, **kwargs)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/api/v1/resources.py", line 239, in post
    **kwargs,
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/api/rest.py", line 971, in initiate_payment
    lock_timeout=lock_timeout,
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/api/python.py", line 998, in transfer_and_wait
    lock_timeout=lock_timeout,
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/api/python.py", line 1087, in transfer_async
    lock_timeout=lock_timeout,
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/raiden_service.py", line 1619, in mediated_transfer_async                                                                                                                         
    self.handle_and_track_state_changes([init_initiator_statechange])
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/raiden_service.py", line 867, in handle_and_track_state_changes                                                                                                                   
    for greenlet in self.handle_state_changes(state_changes):
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/raiden_service.py", line 882, in handle_state_changes                                                                                                                             
    for state_change in state_changes
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/raiden_service.py", line 882, in <listcomp>
    for state_change in state_changes
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/storage/serialization/serializer.py", line 82, in serialize                                                                                                                       
    data = schema.dump(obj)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/schema.py", line 558, in dump
    result = self._serialize(processed_obj, many=many)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/schema.py", line 523, in _serialize
    value = field_obj.serialize(attr_name, obj, accessor=self.get_attribute)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/fields.py", line 328, in serialize
    return self._serialize(value, attr, obj, **kwargs)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/fields.py", line 716, in _serialize
    return [self.inner._serialize(each, attr, obj, **kwargs) for each in value]
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/fields.py", line 716, in <listcomp>
    return [self.inner._serialize(each, attr, obj, **kwargs) for each in value]
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/fields.py", line 583, in _serialize
    return schema.dump(nested_obj, many=many)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/schema.py", line 558, in dump
    result = self._serialize(processed_obj, many=many)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/schema.py", line 523, in _serialize
    value = field_obj.serialize(attr_name, obj, accessor=self.get_attribute)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/fields.py", line 328, in serialize
    return self._serialize(value, attr, obj, **kwargs)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/fields.py", line 1524, in _serialize
    for k in value.keys()
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/marshmallow/fields.py", line 1524, in <dictcomp>
    for k in value.keys()
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/storage/serialization/schemas.py", line 118, in _serialize                                                                                                                        
    return to_hex_address(value)
  File "/home/karl/scenario-player/venv/lib/python3.7/site-packages/raiden/utils/formatting.py", line 81, in to_hex_address
    return AddressHex(HexStr("0x" + address.hex()))
AttributeError: 'str' object has no attribute 'hex'
```

Since it works in the SP CI, I might be doing something wrong.

**Comment by konradkonrad** [comment]
After adding `AttributeError` to https://github.com/raiden-network/raiden/blob/578c95944a55368879a3551ee9753c0943e8e7d0/raiden/storage/serialization/serializer.py#L83 we see
```
Can't serialize: ActionInitInitiator(transfer=TransferDescriptionWithSecretState(amount=1000000000000000, token_network_address=b'\xb9\xcc\xde\xb5rq\xe6\xdb2\x0fu\xe6\x8e:`\xd2OVom', target=b'\x86\x93 \xd8\xc0&\xb4\xc0w\x08\xd6N\xcc\x19/\xdb\xcf\x908\x94', secrethash=b"x)y\xf3jN\xd4\xc0\x8a\xf5\xf7\xa7\x0e\xa2n:\xea\xfa\xac'\xa5o\xb4\x9a)\xc4\xda\xccGx[\xf9", lock_timeout=None), routes=[RouteState (0x2c208b1b37E78bD962A8F2E489B0B73dda0365A3 -> 0x5D433c6baD3fDC12a64c0c4a2de0F68fDBC18684 -> 0x869320d8C026b4c07708D64Ecc192fDBCF903894), fee: 0, RouteState (0x2c208b1b37E78bD962A8F2E489B0B73dda0365A3 -> 0x83d6d0209f521af4A08342F05E22FF33DB4142B8 -> 0x4C2e31328c4D536540cb0249E5B05f512D14ac1e -> 0x869320d8C026b4c07708D64Ecc192fDBCF903894), fee: 0])
```

**Comment by konradkonrad** [comment]
After changing `__repr__` of `RouteState` to `str(self.__dict__)` we see even better:
```
Can't serialize: ActionInitInitiator(transfer=TransferDescriptionWithSecretState(amount=1000000000000000, token_network_address=b'\xb9\xcc\xde\xb5rq\xe6\xdb2\x0fu\xe6\x8e:`\xd2OVom', target=b'\xfb\x17\xbe\x89\x02eP\xa6\xf6FP\xa7\x0f*\xbb\xa6U A5', secrethash=b'T\xa8\xb9\xd9\xd8\xff\x81`\x0b@MFb,?]\x1f\xf5\xdc\xa2\xc8\x9b\xfe\xd8?\x08e%\x90\x96\xebt', lock_timeout=None), routes=[{'route': [b"m\t\xda\x7f_\x10\x02\xe3\xceb'#\xfc\x87\xc6[\xdf\xab*\x8d", b'R:\xd7$4Xz\x96\xc0\x80fV\x85\xe1\xac\x80&A`\xf4', b'\xfb\x17\xbe\x89\x02eP\xa6\xf6FP\xa7\x0f*\xbb\xa6U A5'], 'address_to_metadata': {'0x6d09Da7f5F1002E3ce622723Fc87c65BDFab2A8d': {'user_id': '@0x6d09da7f5f1002e3ce622723fc87c65bdfab2a8d:transport.transport01.raiden.network', 'capabilities': {'Receive': True, 'Mediate': True, 'Delivery': True, 'webRTC': False, 'toDevice': True}}, '0x523ad72434587a96C080665685E1Ac80264160F4': {'user_id': '@0x523ad72434587a96c080665685e1ac80264160f4:transport.transport01.raiden.network', 'capabilities': {'Receive': True, 'Mediate': True, 'Delivery': True, 'webRTC': False, 'toDevice': True}}, '0xfb17bE89026550A6f64650A70f2aBBa655204135': {'user_id': '@0xfb17be89026550a6f64650a70f2abba655204135:transport.transport01.raiden.network', 'capabilities': {'Receive': True, 'Mediate': True, 'Delivery': True, 'webRTC': False, 'toDevice': True}}}, 'swaps': {}, 'estimated_fee': 0}, {'route': [b"m\t\xda\x7f_\x10\x02\xe3\xceb'#\xfc\x87\xc6[\xdf\xab*\x8d", b'\xa5l\x80{oa\xf0\xa3\xd9AIK\x83u\xe2\xbbR3a\x00', b'\xf7\x10O\xca\xceJ\xb2\xd6\x9c\x9b\x04\xd4\xe5\xe4\xca\x0cY.Gf', b'\xfb\x17\xbe\x89\x02eP\xa6\xf6FP\xa7\x0f*\xbb\xa6U A5'], 'address_to_metadata': {'0x6d09Da7f5F1002E3ce622723Fc87c65BDFab2A8d': {'user_id': '@0x6d09da7f5f1002e3ce622723fc87c65bdfab2a8d:transport.transport01.raiden.network', 'capabilities': {'Receive': True, 'Mediate': True, 'Delivery': True, 'webRTC': False, 'toDevice': True}}, '0xA56C807b6F61F0a3D941494B8375e2BB52336100': {'user_id': '@0xa56c807b6f61f0a3d941494b8375e2bb52336100:transport.transport01.raiden.network', 'capabilities': {'Receive': True, 'Mediate': True, 'Delivery': True, 'webRTC': False, 'toDevice': True}}, '0xf7104fcACE4Ab2d69c9B04d4e5E4CA0C592e4766': {'user_id': '@0xf7104fcace4ab2d69c9b04d4e5e4ca0c592e4766:transport.transport01.raiden.network', 'capabilities': {'Receive': True, 'Mediate': True, 'Delivery': True, 'webRTC': False, 'toDevice': True}}, '0xfb17bE89026550A6f64650A70f2aBBa655204135': {'user_id': '@0xfb17be89026550a6f64650a70f2abba655204135:transport.transport01.raiden.network', 'capabilities': {'Receive': True, 'Mediate': True, 'Delivery': True, 'webRTC': False, 'toDevice': True}}}, 'swaps': {}, 'estimated_fee': 0}])
```

In particular the keys of `address_to_metadata` seem to be `str` encoded hex addresses instead of `bytes` encoded raw addresses:
```
'address_to_metadata': {'0x6d09Da7f5F1002E3ce622723Fc87c65BDFab2A8d': {'user_id': '@0x6d09da7f5f1002e3ce622723fc87c65bdfab2a8d:transport.transport01.raiden.network', 'capabilities': {'Receive': True, 'Mediate': True, 'Delivery': True, 'webRTC': False, 'toDevice': True}}, '0x523ad72434587a96C080665685E1Ac80264160F4': [...]
```

Now we need to find out where the encoding originates (server side or when parsing the server response).

**Comment by konradkonrad** [comment]
fixed by #6949
